evap/student/templates/student_vote.html

```diff
@@ -132,6 +132,30 @@ <h3 class="mb-3">{{ course.name }} ({{ course.semester.name }})</h3>
             // (the data gets deleted every time the form is submitted, i.e. also when the form had errors and is displayed again)
             sisyphus.saveAllData();
 
+            // Show a warning on textareas if they only contain obsolete text (punctuation, k.A., n/a, ...)
+            function isTextareaInvalid(textarea) {
+                let val = textarea.val();
+                return /^\W+$/g.test(val) || ["ka", "na"].indexOf(val.toLowerCase().replace(/\W/g,'')) != -1;
+            }
+            function updateTextareaWarning(textarea) {
+                let showWarning = isTextareaInvalid(textarea);
+                textarea.parents('tr').find('.warning-label').toggleClass('d-none', !showWarning);
+                textarea.toggleClass('warning-border', showWarning);
+            }
+            $('textarea').on('input', function() { // Run the check on input
+                let textarea = $(this);
+                let isDelayed = !textarea.hasClass('warning-border') && textarea.val().length < 3 && isTextareaInvalid(textarea)
+                setTimeout(function() {
+                    updateTextareaWarning(textarea);
+                }, isDelayed ? 1000:0);
```

Comment:
i don't quite get the delay logic. what would make sense to me is to immediately do the update if there is already a warning on the textarea, and delay the update otherwise. what do the length and invalid check here do?

also, i think i would do
```
if (theCondition) {
    updateTextareaWarning(textarea);
else {
    setTimeout(... updateTextareaWarning(textarea) ...);
}
```
which matches my mental model of "if that is the case, do it now, otherwise do it a second" more closely, compared to "do this later, and to be precise if that is the case do it in a second and otherwise do it now".

also, you can do `setTimeout( () => updateTextareaWarning(textarea), 1000);` which looks nicer imo.

Reply:
We want to show the warning as early as possible, but not if the user might be just starting or correcting the first entry of a list.

A list is typically started with `- `, so the third character is a letter. If the first 3 letters are punctuation, it's probably not a list and we show the warning.
If the user leaves it at `-`, we show the warning after a delay, because they might be starting a list.
The invalid check is there to prevent that the warning is shown without the right delay when changing the value like the follwing (`|` denoting the users cursor)  `- foo|` --> `- |` --> `- bar|`. The middle state here is invalid and the warning would be almost instantly shown in normal us due to the delayed call being triggered when removing the Os.
As you figured, removing the warning should also be without delay.

Reply:
Now using debounce which should solve all of these concerns...